homeassistant/components/telegram_bot/polling.py

```diff
@@ -55,73 +42,68 @@ def _stop_bot(_event):
     return True
 
 
+@callback
+def process_error(bot, update, error):
+    """Telegram bot error handler."""
+    from telegram.error import (
+        TelegramError, TimedOut, NetworkError, RetryAfter)
+
+    try:
+        raise error
+    except (TimedOut, NetworkError, RetryAfter):
+        # Long polling timeout or connection problem. Nothing serious.
+        pass
+    except TelegramError:
+        _LOGGER.error('Update "%s" caused error "%s"', update, error)
+
+
+def message_handler(handler):
+    """Create messages handler."""
+    from telegram import Update
+    from telegram.ext import Handler
+
+    class MessageHandler(Handler):
+        """Telegram bot message handler."""
+
+        def __init__(self):
+            """Initialize the messages handler instance."""
+            super(MessageHandler, self).__init__(handler)
+
+        def check_update(self, update):
+            """Check is update valid."""
+            return isinstance(update, Update)
+
+        def handle_update(self, update, dispatcher):
+            """Handle update."""
+            optional_args = self.collect_optional_args(dispatcher, update)
+            return self.callback(dispatcher.bot, update, **optional_args)
+
+    return MessageHandler()
+
+
 class TelegramPoll(BaseTelegramBotEntity):
     """Asyncio telegram incoming message handler."""
 
     def __init__(self, bot, hass, allowed_chat_ids):
         """Initialize the polling instance."""
+        from telegram.ext import Updater
+
         BaseTelegramBotEntity.__init__(self, hass, allowed_chat_ids)
-        self.update_id = 0
-        self.websession = async_get_clientsession(hass)
-        self.update_url = '{0}/getUpdates'.format(bot.base_url)
-        self.polling_task = None  # The actual polling task.
-        self.timeout = 15  # async post timeout
-        # Polling timeout should always be less than async post timeout.
-        self.post_data = {'timeout': self.timeout - 5}
+
+        self.updater = Updater(bot=bot, workers=4)
```

Comment:
What is the benefit of 4 workers? If you have 4 bots, you run 16 workers?

Reply:
It's default value in [python-telegram-bot updater](https://github.com/python-telegram-bot/python-telegram-bot/blob/439790375ed8ed493c43e464aa8e2b60a77939db/telegram/ext/updater.py#L94)

Reply:
In that case, remove the default value if they will be default